### DISCUSSION TITLE
import MySQLdb doesn't get cursors

### DISCUSSION THREAD

**Discussion by jmath222** [PyMySQL/mysqlclient#559] [General]
Anyone know why this happens? Cursors is in the `__all__` in the `__init__.py`, so it should be available after import.

$ python
Python 3.8.8 (default, Aug 11 2021, 06:52:42)
[GCC 8.5.0 20210514 (Red Hat 8.5.0-3)] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import MySQLdb
>>> MySQLdb.cursors
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
AttributeError: module 'MySQLdb' has no attribute 'cursors'

**Comment by methane** [comment]
> Cursors is in the `__all__` in the `__init__.py`, so it should be available after import.

You misunderstood `__all__`. It isn't automatic import.

**Reply by jmath222** [reply]
Understood. I see that using `from MySQLdb import *` would get cursors.

I don't want to suggest a change to the package, but want to explain why I brought this up. My organization experienced a difficult to debug import issue because we (and PyCharm) assumed `import MySQLdb` also imported `MySQLdb.cursors`

We had working module A, which only imported MySQLdb but used MySQLdb.cursors.
Module B imported MySQLdb.cursors, but no longer used it after a change.
Module A imported module B, which brought MySQLdb.cursors into the MySQLdb namespace in module A.

A developer was cleaning up module B and removed the MySQLdb.cursors import, as PyCharm correctly marked the import as unused. This broke module A, as `cursors` was no longer available in its MySQLdb namespace.

This doesn't sound like the fault of MySQLdb itself, maybe just an idiosyncrasy of python's import system/PyCharm. However I can't remember ever having an issue like this with another package so I wanted to point it out.
